metronome/checkpointing/models/src/io/iohk/metronome/checkpointing/models/ETCBlock.scala

```diff
@@ -0,0 +1,36 @@
+package io.iohk.metronome.checkpointing.models
+
+import io.iohk.metronome.crypto.hash.Hash
+
+/** ETCBlock is a simpler version of original ETC block in the chain that
+  * supports the running of the checkpointing interpreter.
+  */
+case class ETCBlock(
+    header: ETCBlock.Header,
+    body: ETCBlock.Body
+) {
+  def hash: Hash = header.hash
+}
```

Comment:
ETC already has its own block definition. I don't think we should see chain specific stuff in `checkpointing.models`. Whaver is required to implement the interpreter should live in https://github.com/input-output-hk/mantis

Only the most generic code should be added to the `checkpointing` module that works with everything. I don't want it to end up as a union of all possible chains we can deal with.